src/lib/select/select.md

```diff
@@ -1,111 +1,161 @@
 
-If you want one of your options to reset the select's value, you can omit specifying its value:
+If you want one of your options to reset the select's value, you can omit specifying its value.
 
-*my-comp.html*
-```html
-<mat-form-field>
-  <mat-select placeholder="State">
-    <mat-option>None</mat-option>
-    <mat-option *ngFor="let state of states" [value]="state.code">{{ state.name }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
+<!-- example(select-reset) -->
 
-### Setting a static placeholder
+### Creating groups of options
 
-It's possible to turn off the placeholder's floating animation using the `floatPlaceholder` property. It accepts one of three string options:
-- `'auto'`: This is the default floating placeholder animation. It will float up when a selection is made.
-- `'never'`: This makes the placeholder static. Rather than floating, it will disappear once a selection is made.
-- `'always'`: This makes the placeholder permanently float above the input. It will not animate up or down.
+The `<mat-optgroup>` element can be used to group common options under a subheading. The name of the
+group can be set using the `label` property of `<mat-optgroup>`. Like individual `<mat-option>`
+elements, an entire `<mat-optgroup>` can be disabled or enabled by setting the `disabled` property
+on the group.
 
-```html
-<mat-form-field floatPlaceholder="never">
-  <mat-select placeholder="State" [(ngModel)]="myState">
-  <mat-option *ngFor="let state of states" [value]="state.code">{{ state.name }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
+<!-- example(select-optgroup) -->
+
+### Selecting multiple options
+
+By default the `<mat-select>` only supports selecting a single option at a time. However you can use
+the `multiple` property to allow the user to select multiple values at once. When using the
+`<mat-select>` in multiple selection mode, its value will be a sorted list of all selected values
+rather than a single value.
+
+<!-- example(select-multiple) -->
+
+### Customizing the trigger label
+
+If you want to display a custom trigger label inside a select, you can use the
+`<mat-select-trigger>` element.
+
+<!-- example(select-custom-trigger) -->
+
+### Disabling the ripple effect
 
-Global default placeholder options can be specified by setting the `MAT_PLACEHOLDER_GLOBAL_OPTIONS` provider. This setting will apply to all components that support the floating placeholder.
+By default when a user clicks on a `<mat-option>` a ripple animation is shown. This can be disabled
+by setting the `disableRipple` property on `<mat-select>`.
+
+<!-- example(select-no-ripple) -->
+
+### Adding custom styles to the dropdown panel
+
+In order to facilitate easily styling the dropdown panel, `<mat-select>` has a `panelClass` property
+which can be used to apply additional CSS classes to the dropdown panel.
+
+<!-- example(select-panel-class) -->
+
+### Changing when error messages are shown
+
+The `<mat-form-field>` allows you to
```

Comment:
A few of these sections are going to be the same between the select and input. Maybe we should consider (not necessarily now) having a common page for them?

Reply:
Yes, I'd also like to consolidate the code itself so we're not duplicating it in input and select, just need to find a good way to make it work with change detection